### ISSUE TITLE
Speed up vis_drake_graph() on large completed workflows

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#1098]
## Prework

- [x] Read and abide by `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CODE_OF_CONDUCT.md).
- [x] Search for duplicates among the [existing issues](https://github.com/ropensci/drake/issues), both open and closed.
- [x] Advanced users: verify that the bottleneck still persists in the current development version (i.e. `remotes::install_github("ropensci/drake")`) and mention the [SHA-1 hash](https://git-scm.com/book/en/v1/Getting-Started-Git-Basics#Git-Has-Integrity) of the [Git commit you install](https://github.com/ropensci/drake/commits/master).

## Description

`build_times()`, `outdated()`, and `progress()` are slowing down `vis_drake_graph()` for completed workflows, and I think I know how to make them super fast. Instead of individual interactions with the cache to get the metadata, we can use vectorized operations like `config$cache$list(namespace = "meta")` and `config$cache$mget(..., namespace = "meta")` to get all the old metadata at once.

## Reproducible example

```r
library(drake)
library(fs)
library(profile)
library(withr)

profile <- function(plan) {
  local_dir(dir_create(tempfile()))
  rprof <- "prof.rprof"
  pprof <- "prof.pprof"
  plan <- drake_plan(
    z = target(w, transform = map(w = !!seq_len(1e3)))
  )
  make(plan)
  config <- drake_config(plan)
  Rprof(filename = rprof)
  tmp <- vis_drake_graph(config)
  Rprof(NULL)
  data <- read_rprof(rprof)
  write_pprof(data, pprof)
  vis_pprof(pprof)
}

vis_pprof <- function(path, host = "localhost", port = NULL) {
  server <- sprintf("%s:%s", host, port %||% random_port())
  message("local pprof server: http://", server)
  args <- c("-http", server, path)
  if (on_windows()) {
    shell(paste(c("pprof", args), collapse = " "))
  } else {
    system2(jointprof::find_pprof(), args)
  }
}

random_port <- function(from = 49152L, to = 65355L) {
  sample(seq.int(from = from, to = to, by = 1L), size = 1L)
}

on_windows <- function() {
  tolower(Sys.info()["sysname"]) == "windows"
}

`%||%` <- function(x, y) {
  if (is.null(x) || length(x) <= 0) {
    y
  } else {
    x
  }
}

profile()
```

## Benchmarks

![Capture](https://user-images.githubusercontent.com/1580860/70383972-3b817880-1945-11ea-90e9-f819e973834e.PNG)

**Comment by wlandau** [comment]
The graph above was from Windows 10. The results for Linux on the same hardware are similar. See below.

![Screenshot_20191207_230548](https://user-images.githubusercontent.com/1580860/70384023-20633880-1946-11ea-8aec-cc4d88a1fbad.png)

**Comment by wlandau** [comment]
Here is the flame graph after https://github.com/ropensci/drake/commit/d3063d736c17762c0ba12b158e8b1ffc8dae76b9.

![Screenshot_20191208_090908](https://user-images.githubusercontent.com/1580860/70390625-97c2b780-199a-11ea-9d04-f8bad8fbf095.png)

Turns out `mget()` is not really appropriate for getting old metadata in `outdated()` since we do not know how much metadata we need to read until we have already advanced through the graph. I did implement a speedup in `outdated()`, but it was minor.
